src/main/java/pl/edu/pw/mini/msi/knowledgerepresentation/ActionLanguageListener.java

```diff
@@ -0,0 +1,395 @@
+package pl.edu.pw.mini.msi.knowledgerepresentation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.edu.pw.mini.msi.knowledgerepresentation.actionDomain.ActionDomain;
+import pl.edu.pw.mini.msi.knowledgerepresentation.actionDomain.sentenceParts.*;
+import pl.edu.pw.mini.msi.knowledgerepresentation.actionDomain.sentences.*;
+import pl.edu.pw.mini.msi.knowledgerepresentation.grammar.ActionLanguageBaseListener;
+import pl.edu.pw.mini.msi.knowledgerepresentation.grammar.ActionLanguageParser;
+
+public class ActionLanguageListener extends ActionLanguageBaseListener {
+
+    private static final Logger log = LoggerFactory.getLogger(ActionLanguageListener.class);
+
+    private Action lastAction = null;
+    private Action preLastAction = null;
+    private Actor lastActor = null;
+    //private ActionLanguageParser.LogicalExpressionContext lastLogExpr = null;
+    private EQueryType lastQueryType = EQueryType.always;
+    private Fluent lastFluent = null;
+    private IFormula lastFormula = null;
+    private ScenarioName lastScenarioName = null;
+    private Time lastTime = null;
+    private boolean lastTypically = false;
+
+
+    private boolean isInLogicalExpression = false;
+    private IFormula lastUnderConditionFormula = null;
+    private boolean isInUnderCondition = false;
+
+    //===========================================================================================
+    private ActionDomain actionDomain = new ActionDomain();
+
+
+    public ActionLanguageListener(ActionDomain actionDomain) {
+        this.actionDomain = actionDomain;
+        log.debug("ActionLanguageListener(ActionDomain actionDomain)");
+    }
+
+    public ActionDomain getActionDomain() {
+        //actionDomain.baseSentences.add(...);
+        //actionDomain.scenarios.add(...);
+        //actionDomain.addQueries(...);
+        actionDomain.calculateFullScenarios();
+        actionDomain.calculateMappedQueries();
+        actionDomain.fillFluentsAndActionsIDs();
+
+        return actionDomain;
+    }
+    //ActionLanguageParser.LogicalExpressionContext ctx;
+    //===========================================================================================
+
+    /**
+     * {@inheritDoc}
+     * <p/>
+     * <p>The default implementation does nothing.</p>
+     */
+    @Override
+    public void enterProgramm(ActionLanguageParser.ProgrammContext ctx) {
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p/>
+     * <p>The default implementation does nothing.</p>
+     */
+    @Override
+    public void exitProgramm(ActionLanguageParser.ProgrammContext ctx) {
+    }
+
+    private void clear(boolean clearLastScenarioName) {
+        lastAction = null;
+        preLastAction = null;
+        lastActor = null;
+        //ActionLanguageParser.LogicalExpressionContext lastLogExpr = null;
+        lastQueryType = null;

```

Comment:
\[Checkstyle\] WARNING: 'method def' child have incorrect indentation level 8, expected level should be 4\.